pkg/queue/instance_test.go

```diff
@@ -101,13 +101,13 @@ func TestResourceFree(t *testing.T) {
 		return nil
 	})
 
-	// mock queue wait cond signal
-	time.Sleep(200 * time.Millisecond)
-
-	close(stop)
+	// mock queue block wait cond signal
+	time.AfterFunc(10*time.Millisecond, func() {
+		close(stop)
+	})
 
 	select {
-	case <-time.After(500 * time.Microsecond):
+	case <-time.After(20 * time.Millisecond):
```

Comment:
can you make the timeout higher? with 20ms `12627 runs so far, 3 failures` with 200ms `30932 runs so far, 0 failures` using https://godoc.org/golang.org/x/tools/cmd/stress

Reply:
thanks, in my local get `39438 runs so far, 0 failures`